### ISSUE TITLE
Syntax for lexer keywords and all subclasses

### ISSUE THREAD

**Issue by ba205** [kach/nearley#439]
I'm writing a parser for a small language. In my lexer, I have something like the following:

```
identifier: {match: /[a-zA-Z_0-9]+/, type: moo.keywords({
         'keyword-class-one' : <...>,
         'keyword-class-two' : <...>,
         ...
         'keyword-class-n' : <...>,
    })}
```

This does a good job of performing a longest case match on identifiers with many keyword classes. Usually, I can use %identifiers and %keyword-class-i and my parser will work just fine.

However, comments in the language can have keywords in them, and right now I use '%identifier' to denote comments, which works fine except when the comment can actually be a keyword class. Is there an easier way to do this other than '(%identifier|%keyword-class-one|...|%keyword-class-n)'? Like saying "select this lexed identifier and all of it's subclasses"?

**Comment by kach** [comment]
@tjvr can speak more to this, but I think you want your comments to get matched as a special "comment" token (or even better, simply ignored) rather than matching as identifiers.

**Comment by ba205** [comment]
I attempted an approach using this method, and came up with the following regex to match comments: 

```^[%:@][a-zA-Z_0-9]+```

This is the correct description of comments in my language, and I tested the regex here: https://regex101.com. However, when put in my lexer, it breaks previously working cases of comments when I handled ```[%,:,@] ``` separately from the string/number catch-all, and concatenated the results in post processing.

I noticed in a previous issue that full regex isn't supported. But I'm curious, is there a particular reason why this regex wouldn't be good in a lexer?
